File: tests/rosetta/transpiler/Kotlin/ackermann-function-2.kt
fun pow(base: Int, exp: Int): Int {
    var result: Int = 1
    var i: Int = 0
    while (i < exp) {
        result = result * base
        i = i + 1
    }
    return result
}

fun ackermann2(m: Int, n: Int): Int {
    if (m == 0) {
        return n + 1
    }
    if (m == 1) {
        return n + 2
    }
    if (m == 2) {
        return (2 * n) + 3
    }
    if (m == 3) {
        return (8 * pow(2, n) as Int) - 3
    }
    if (n == 0) {
        return ackermann2(m - 1, 1) as Int
    }
    return ackermann2(m - 1, ackermann2(m, n - 1) as Int) as Int
}

fun user_main(): Unit {
    println("A(0, 0) = " + ackermann2(0, 0) as Int.toString())
    println("A(1, 2) = " + ackermann2(1, 2) as Int.toString())
    println("A(2, 4) = " + ackermann2(2, 4) as Int.toString())
    println("A(3, 4) = " + ackermann2(3, 4) as Int.toString())
}

fun main() {
    user_main()
}

File: tests/rosetta/transpiler/php/ackermann-function-2.php
<?php
ini_set('memory_limit','-1');
function mochi_pow($base, $exp) {
  global $ackermann2, $main;
  $result = 1;
  $i = 0;
  while ($i < $exp) {
  $result = $result * $base;
  $i = $i + 1;
};
  return $result;
}
function ackermann2($m, $n) {
  global $mochi_pow, $main;
  if ($m == 0) {
  return $n + 1;
}
  if ($m == 1) {
  return $n + 2;
}
  if ($m == 2) {
  return 2 * $n + 3;
}
  if ($m == 3) {
  return 8 * mochi_pow(2, $n) - 3;
}
  if ($n == 0) {
  return ackermann2($m - 1, 1);
}
  return ackermann2($m - 1, ackermann2($m, $n - 1));
}
function main() {
  global $mochi_pow, $ackermann2;
  echo "A(0, 0) = " . json_encode(ackermann2(0, 0), 1344), PHP_EOL;
  echo "A(1, 2) = " . json_encode(ackermann2(1, 2), 1344), PHP_EOL;
  echo "A(2, 4) = " . json_encode(ackermann2(2, 4), 1344), PHP_EOL;
  echo "A(3, 4) = " . json_encode(ackermann2(3, 4), 1344), PHP_EOL;
}
main();
